### PR TITLE
OTLP tracing

### PR DESCRIPTION
**Pull request by mystenmark**
Enables export of tracing span data via OTLP. Can be viewed with grafana/tempo: 

image

Commits:
- OTLP tracing
- Enable tracing via admin server
- Local grafana instance via docker
- Add tracing to execution path
- Update docs
- Update Cargo.lock

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -12392,6 +12495,34 @@ dependencies = [
  "tracing-futures",
 ]
 
+[[package]]
+name = "tonic"
+version = "0.9.2"
```
Comment (bmwill) [inline Cargo.lock:12500]: maybe we should look at what it would take to update tonic so we don't have duplicate deps here

File: crates/sui-node/src/admin.rs [hunk 2]
```diff
@@ -87,8 +92,33 @@ pub async fn run_admin_server(node: Arc<SuiNode>, port: u16, filter_handle: Filt
         .unwrap()
 }
 
+#[derive(Deserialize)]
+struct EnableTracing {
+    filter: String,
+    duration: String,
+}
```
Comment (bmwill) [inline crates/sui-node/src/admin.rs:99]: Is there a way to disable it too?

File: crates/sui-node/src/main.rs [hunk 3]
```diff
@@ -73,6 +71,8 @@ fn main() {
         .with_prom_registry(&prometheus_registry)
         .init();
 
+    drop(metrics_rt);
+
```
Comment (bmwill) [inline crates/sui-node/src/main.rs:75]: Are these changes necessary?

File: crates/sui-swarm/src/memory/container.rs [hunk 4]
```diff
@@ -44,12 +45,21 @@ impl Container {
...
-                tracing::Level::INFO,
-                "node",
-                name =% AuthorityPublicKeyBytes::from(config.protocol_key_pair().public()).concise(),
-            );
-            let _guard = span.enter();
+            let span = if get_global_telemetry_config()
+                .map(|c| c.enable_otlp_tracing)
+                .unwrap_or(false)
+            {
+                // we cannot have long-lived root spans when exporting trace data to otlp
```
Comment (bmwill) [inline crates/sui-swarm/src/memory/container.rs:52]: Whats the rationale behind this? It would be sad to loose this information

Comment (mystenmark) [reply crates/sui-swarm/src/memory/container.rs:52]: These spans only exist to provide the per-node context in the logging output. But when exporting this as OTLP it really screws up the viewers. Spans don't get exported until they end, and these spans never end, so every other span is missing its root.

This doesn't really matter since this is only an issue when tracing `sui` or `sui-test-validator`, so I could revert this change. It was just convenient to be able to use `sui-test-validator` for testing this.

File: crates/sui-node/src/main.rs (hunk 3 above)
Comment (mystenmark) [reply crates/sui-node/src/main.rs:75]: Yes - otherwise when we initialize TelemetryConfig, there is no active tokio runtime, so it can't install the batch exporter.

File: crates/sui-node/src/admin.rs (hunk 2 above)
Comment (mystenmark) [reply crates/sui-node/src/admin.rs:99]: It disables automatically after the duration expires. I guess it wouldn't hurt to be able to disable it actively in case you set too long a duration by accident.

File: Cargo.lock (hunk 1 above)
Comment (mystenmark) [reply Cargo.lock:12500]: Upgraded us to 0.10, hopefully otel follows suit sometime soon.

File: crates/sui-node/src/admin.rs (hunk 2 above)
Comment (mystenmark) [reply crates/sui-node/src/admin.rs:105]: added a disable route

**Review by bmwill — APPROVED** [review]
